src/basic/in-addr-util.c

```diff
@@ -485,6 +485,36 @@ int in_addr_ifindex_from_string_auto(const char *s, int *family, union in_addr_u
         return r;
 }
 
+int in_addr_ifindex_name_from_string_auto(const char *s, int *family, union in_addr_union *ret, int *ifindex, char **server_name) {
```

Comment:
hmm, i don't thin this should be in src/basic/, it's too specific to resolved. Can you move it to src/resolve/ at a good place?